JetStreamCompose/jetstream/src/main/java/com/google/jetstream/JetStreamApplication.kt

```diff
@@ -33,9 +37,20 @@ class JetStreamApplication : Application()
 @Module
 object MovieRepositoryModule {
 
+    @Singleton
```

Comment:
For optimal implementation, here's what you should do:

1. Use constructor injection for `MovieRepositoryImpl`

```
@Singleton
class MovieRepositoryImpl(
    movieDataSource: MovieDataSource,
    tvDataSource: TvDataSource,
    movieCastDataSource: MovieCastDataSource,
    movieCategoryDataSource: MovieCategoryDataSource
)
```

2. Bind the `MovieRepositoryImpl` to `MovieRepository`

```
@Binds
abstract fun bindMovieRepository(
  movieRepositoryImpl: MovieRepositoryImpl
): MovieRepository
``` 

https://developer.android.com/training/dependency-injection/hilt-android#inject-interfaces

Reply:
Thanks. Updated as suggested, while I could not find any explicit description why the Injection with @Binds is more optimal than using @Provides.